language/stdlib/modules/shared_ed25519_public_key.move

```diff
@@ -34,6 +35,11 @@ module SharedEd25519PublicKey {
 
     fun rotate_key(shared_key: &mut T, new_public_key: vector<u8>) {
         Transaction::assert(Vector::length(&new_public_key) == 32, 7000);
```

Comment:
Can probably kill this check now (since `validate_pubkey` presumably returns false for a bad length)?

Reply:
The nice thing you have with this is the consistent error code for a length error.